src/Serializer/ItemNormalizer.php

```diff
@@ -37,11 +39,17 @@ public function normalize($object, $format = null, array $context = [])
 
     /**
      * {@inheritdoc}
+     *
+     * @throws InvalidArgumentException
      */
     public function denormalize($data, $class, $format = null, array $context = [])
     {
         // Avoid issues with proxies if we populated the object
         if (isset($data['id']) && !isset($context['object_to_populate'])) {
+            if (isset($context['allow_update']) && true !== $context['allow_update']) {
```

Comment:
> Still, the allow_update from api_platform will override the previous ones no? So it's our behavior that will still be taken into consideration, won't it?

Yes, but it'll break the semantics defined elsewhere (so our code will work correctly but some other code will break - not good at all). That's the whole problem with naming conflicts.

> IMO it's too late to change existing keys and it will make the API less obvious for newcomers. And it's unlikely that Symfony add a allow_update key in its built-in normalizers.

I don't think it's too late, now is the last chance to do it. Even if Symfony Serializer does not add it, it still has the potential to conflict with user-defined keys. It's okay for Symfony Serializer itself to use unprefixed keys, because that's just something consumers will have to cope with. I don't think it's good for us to do that though... (Imagine other packages / bundles doing the same, it'll end up in conflict eventually.)